Check 'auth' field when displaying .dockercfg config

jhadvig opened the pull request:
While working on https://bugzilla.redhat.com/show_bug.cgi?id=1506998 I've noticed that we are not checking for `auth` field in the old **.dockercfg** format, which looks like:
```
{
        "registryaddress": {
                "auth": "auth string",
                "email": "email@domain.com"
        }
}
```

The `registryaddress` could also contain `username` and `password` fields.

@spadgett PTAL

Will add the dist after review

spadgett commented on app/scripts/services/secrets.js:
```diff
@@ -28,17 +28,29 @@ angular.module("openshiftConsole")
       return secretsByType;
     };
 
+    var setDataParams = function(decodedSecretData, serverName, data) {
```
If this is specific to dockercfg, we should have that somewhere in the function name.

jhadvig replied:
This method is not intended to be specific for `dockercfg` or `config.json`. It's purpose is to assemble an object that will contain server address, username, password.

spadgett commented on app/scripts/services/secrets.js:
```diff
@@ -28,17 +28,29 @@ angular.module("openshiftConsole")
+        username: usernamePassword[0],
+        password: usernamePassword[1],
+        email: data.email
+      };
```
I would let the caller set the value in the auths object and just let this function return the data. Then the function takes fewer arguments and doesn't need to know anything about the structure of the `decodedSecretData` object passed to it.

(untested)

```js
var getDockercfgServerParams = function(serverData) {
  var params = _.pick(serverData, ['email', 'username', 'password']);
  if (serverData.auth) {
    try {
      // Decode Base64-encoded username:password string.
      _.spread(window.atob(data.auth).split(":"), function(username, password) {
        params.username = username;
        params.password = password;
      });
    } catch(e) {}
  }

  return params;
};
```

then below in the `decodeDockercfg` function

```js
decodedSecretData.auths[serverName] = getDockercfgServerParams(data);
```

https://lodash.com/docs/4.17.4#pick
https://lodash.com/docs/4.17.4#spread

spadgett commented on app/scripts/services/secrets.js:
```diff
@@ -28,17 +28,29 @@ angular.module("openshiftConsole")
+    var setDataParams = function(decodedSecretData, serverName, data) {
+      var usernamePassword = [];
+      if (data.auth) {
+        usernamePassword = window.atob(data.auth).split(":");
```
We should catch exceptions to avoid a runtime error if it's not properly encoded.

https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/atob

jhadvig commented:
@spadgett I've updated the PR based on your comments. 
I know that you suggested to rename the method for getting the server params to `getDockercfgServerParams` but that method could be Docker config format agnostic (should cover both `.dockercfg` and `docker/config.json`).
Also I've refactored both `decodeDockercfg` and `decodeDockerconfigjson` method into single `decodeDockerConfig` method since both of them have been doing almost the same operations, there have been just slight differences which should  be covered with a simple `if`.
Also added decoding error handling, here is the example with error notification:
![auth_err](https://user-images.githubusercontent.com/1668218/32176983-5be0eec2-bd89-11e7-89f8-a722375acb16.png)


Tested the change with single and multi server `.dockercfg` and `docker/config.json` and works as expected.

PTAL

spadgett commented:
/kind bug

@jhadvig You have a dist mismatch

spadgett commented:
/lgtm

spadgett commented:
Hm.

```
pm ERR! phantomjs-prebuilt@2.1.15 install: `node install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the phantomjs-prebuilt@2.1.15 install script 'node install.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the phantomjs-prebuilt package,
npm ERR! not with npm itself.
```

/retest
